Framework/API/inc/MantidAPI/AlgorithmProperty.h

```diff
@@ -34,16 +36,18 @@ API::IAlgorithm interface.
 @date 24/03/2011
 */
 class MANTID_API_DLL AlgorithmProperty
-    : public Kernel::PropertyWithValue<boost::shared_ptr<IAlgorithm>> {
+    : public Kernel::PropertyWithValue<std::shared_ptr<IAlgorithm>> {
 public:
   /// Typedef the held type
-  using HeldType = boost::shared_ptr<IAlgorithm>;
+  using HeldType = std::shared_ptr<IAlgorithm>;
 
   /// Constructor
   AlgorithmProperty(const std::string &propName,
                     Kernel::IValidator_sptr validator =
                         Kernel::IValidator_sptr(new Kernel::NullValidator),
                     unsigned int direction = Kernel::Direction::Input);
+
+  AlgorithmProperty(const AlgorithmProperty &);
```

Comment:
Can we not `= default` this?

Similar for most of the rest that have been defined but just do a member-wise copy?

Reply:
Done: I've gone ahead and removed any `default` for the rule of 1 where possible, or stuck to the rule of 3 where they were protected for example